src/mono/Directory.Build.props

```diff
@@ -1,25 +1,17 @@
 <Project>
   <PropertyGroup>
-    <SkipImportArcadeSdkFromRoot>true</SkipImportArcadeSdkFromRoot>
-  </PropertyGroup>
-  <Import Project="..\..\Directory.Build.props" />
```

Comment:
If we replace five occurrences of `HostArch` with `BuildArchitecture` in `src/mono/llvm/llvm-init.proj`, I think we can then simplify the top of this file as follows; to match the properties evaluation of master branch:

```xml
<Project>
  <Import Project="..\..\Directory.Build.props" />

  <PropertyGroup>
    <!-- Set default Platform -->
    <Platform>$(TargetArchitecture)</Platform>
    <BaseIntermediateOutputPath>$([MSBuild]::NormalizeDirectory('$(ArtifactsObjDir)', 'mono', '$(MSBuildProjectName)'))</BaseIntermediateOutputPath>
    ...
```
as there is no dependency on `$(Platform)` in root `Directory.Build.props` import, and `Condition="'$(TargetArchitecture)' == ''"` never satisfies in the master branch because Configurations.props (which is indirectly imported) has final fallback on x64 https://github.com/dotnet/runtime/blob/e0a431b6e1c066a9fe4684379c904d6c735b6794/eng/Configurations.props#L116

Reply:
Thanks, I cleaned this already up. Platform is needed to be defined before Arcade is imported as Arcade relies on it: https://github.com/dotnet/arcade/blob/7e0f9bc02f83d0a561d9edb963d856334f048f0d/src/Microsoft.DotNet.Arcade.Sdk/tools/RepoLayout.props#L11 for constructing the paths: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectLayout.props.

There is further clean-up possible but that can be done in a subsequent PR :)

Reply:
Ah, ok. So here we rely on `TargetArchitecture` for `Platform` that was passed via the command line, and not the one that will be set by Configurations.props. If value of `/p:TargetArchitecture=..` coming from command line is empty, `Platform` will be set to `AnyCPU` in arcade. 👍

Reply:
Thanks for reminding me that these variables are defined before Configurations.props is imported. I changed the code path so that the previous behavior is preserved. Both  code paths in coreclr and mono should be cleaned up further to make sure that:
1. The intermediate and output paths are same except of the bin/obj difference.
2. Arcade should be the source of truth that defines these paths.

Reply:
Thanks. It has refreshed my knowledge about new arcade import mechanism and its future. :)

Just a thought..
If Configurations.props does not rely on anything from arcade, we can switch the order of imports in root level `Directory.Build.props` (i.e. `Configuration.props` is imported before arcade's `Sdk.props`), and set `<Platform Condition="'$(SkipInferPlatformInConfigurationsProps)' != 'true'">$(TargetArchitecture)</Platform>` in `Configurations.props` for single place of defining all architecture related properties (with proper fallback to x64). This `<SkipInferPlatformInConfigurationsProps>true</SkipInferPlatformInConfigurationsProps>`  will be only set in libraries' `Directory.Build.props` (similar to `SkipInferTargetOSName`), as we want arcade to set AnyCPU for libraries.

Reply:
Interesting thought. What worries me about this a bit is that I would like to keep the properties defined before Arcade is imported at a bare minimum. At some point we don't need the Configurations.props file anymore and can just inline it back into Directory.Build.props. When we do that, we might want to follow your approach and front load the properties that Arcade relies on in their props files.